manager/controllers/baseline_create.go

```diff
@@ -141,29 +150,33 @@ func buildCreateBaselineQuery(db *gorm.DB, request CreateBaselineRequest, accoun
 	return baseline.ID, err
 }
 
-func checkInventoryIDs(db *gorm.DB, accountID int, inventoryIDs []string) (missingIDs []string, err error) {
-	var containingIDs []string
+func checkInventoryIDs(db *gorm.DB, accountID int, inventoryIDs []string,
+) (missingIDs, satelliteManagedIDs []string, err error) {
+	var containingSystems []SystemBaselineDBLookup
 	err = db.Table("system_platform sp").
 		Where("rh_account_id = ? AND inventory_id::text IN (?)", accountID, inventoryIDs).
-		Pluck("sp.inventory_id", &containingIDs).Error
+		Scan(&containingSystems).Error
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	if len(inventoryIDs) == len(containingIDs) {
-		return []string{}, nil // all inventoryIDs found in database
-	}
+	containingIDsMap := make(map[string]bool, len(containingSystems))
+	for _, containingSystem := range containingSystems {
+		containingIDsMap[containingSystem.InventoryID] = true
 
-	containingIDsMap := make(map[string]bool, len(containingIDs))
-	for _, containingID := range containingIDs {
-		containingIDsMap[containingID] = true
+		if containingSystem.SatelliteManaged {
+			satelliteManagedIDs = append(satelliteManagedIDs, containingSystem.InventoryID)
+		}
 	}
 
 	for _, inventoryID := range inventoryIDs {
 		if _, ok := containingIDsMap[inventoryID]; !ok {
 			missingIDs = append(missingIDs, inventoryID)
 		}
 	}
+
 	sort.Strings(missingIDs)
-	return missingIDs, nil
+	sort.Strings(satelliteManagedIDs)
```

Comment:
it seems unnecessary to sort (and even create) `satelliteManagedIDs` slice, I think it would be enough to return `bool` if there is at least 1 satellite system in the list.